lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -651,6 +645,31 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
         parameters
       end
 
+      def add_3ds_info(parameters, three_d_secure_opts)
+        return if empty?(three_d_secure_opts)
+
+        parameters[:three_d_secure_pass_thru] = {
+          three_d_secure_version: three_d_secure_opts[:version],
+          eci_flag: three_d_secure_opts[:eci],
+          cavv_algorithm: three_d_secure_opts[:cavv_algorithm],
+          cavv: three_d_secure_opts[:cavv],
+          directory_response: three_d_secure_opts[:directory_response_status],
+          authentication_response: three_d_secure_opts[:authentication_response_status],
+        }.merge(xid_or_ds_trans_id(three_d_secure_opts))
```

Comment:
What happens here for any blank fields?

Reply:
Good Q. Just tested that and interestingly, it sends the empty elements (`<directory-response nil="true"/>`). The sandbox doesn't complain, but I'll add conditionals here.

Reply:
Updated, including adding tests for sending only part of the `three_d_secure_opts`.